packages/react-conformance/src/defaultTests.tsx

```diff
@@ -65,12 +65,95 @@ export const defaultTests: TestObject = {
   },
 
   /** Constructor/component name matches filename */
-  'name-matches-filename': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
+  'name-matches-filename': (componentInfo: ComponentDoc, testInfo: IsConformantOptions) => {
     it(`Component/constructor name matches filename`, () => {
       const { componentPath, displayName } = testInfo;
       const fileName = path.basename(componentPath, path.extname(componentPath));
 
       expect(displayName).toEqual(fileName);
     });
   },
+
+  /** Ensures component is exported at top level allowing import { Component } from 'packageName' */
+  'exported-top-level': (componentInfo: ComponentDoc, testInfo: IsConformantOptions) => {
+    if (!testInfo.isInternal) {
+      it(`is exported at top-level`, () => {
+        const { displayName, componentPath, Component } = testInfo;
+        const rootPath = componentPath.replace(/src.*/, '');
+        const indexFile = require(path.join(rootPath, 'src', 'index'));
+
+        expect(indexFile[displayName]).toBe(Component);
+      });
+    }
+  },
+
+  /** Ensures component has top level file in package/src/componentName */
+  'has-top-level-file': (componentInfo: ComponentDoc, testInfo: IsConformantOptions) => {
+    if (!testInfo.isInternal) {
+      it(`has corresponding top-level file 'package/src/Component'`, () => {
+        const { displayName, componentPath, Component } = testInfo;
+        const rootPath = componentPath.replace(/src.*/, '');
+        const topLevelFile = require(path.join(rootPath, 'src', displayName));
+
+        expect(topLevelFile.default || topLevelFile[displayName]).toBe(Component);
+      });
+    }
+  },
+
+  /** If the component is a subcomponent, ensure its parent has the subcomponent as static property */
+  'is-static-property-of-parent': (componentInfo: ComponentDoc, testInfo: IsConformantOptions) => {
+    it(`is a static property of its parent`, () => {
+      const { componentPath, displayName, useDefaultExport } = testInfo;
```

Comment:
This is one case where it might be better to do part of the setup (the `isParent` calculation) before declaring the `it()`--this test ideally shouldn't run if the component is a parent.